app/commands/solution/update_num_loc.rb

```diff
@@ -3,11 +3,23 @@ class Solution::UpdateNumLoc
 
   initialize_with :solution
 
-  def call = solution.update!(num_loc:)
+  def call
+    return if num_loc == solution.num_loc
+
+    solution.update!(num_loc:)
+    resync_representation_to_search_index!
+  end
 
   private
+  memoize
   def num_loc
     iteration = solution.latest_published_iteration || solution.latest_iteration
     iteration&.num_loc.to_i
   end
+
+  def resync_representation_to_search_index!
+    Exercise::Representation.where(oldest_solution: solution).each do |representation|
```

Comment:
I probably could have used `find_by` here. Do you want me to do that?

Reply:
Yeah, I considered this when I first reviewed. I think what you have is safer, just in case there are multiple with `oldest_solution` as the same (there shouldn't be, but 🤷)